content/blog/esc-sync-with-iac/index.md

```diff
@@ -0,0 +1,176 @@
+meta_image: "meta.png"
+authors:
+  - komal-ali
+tags:
+  - esc
+  - secrets
+---
+
+Managing secrets across multi-cloud infrastructures has long been a challenge for developers and operations teams. This article explores an IaC-based strategy to centrally define secrets and configuration in ESC and automatically sync these values across the external platforms where they will be utilized, effectively reducing secret sprawl and manual overhead.
+
+<!--more-->
+
+## The Secret Management Conundrum
+
+Traditional secret management often leads to a phenomenon known as secret sprawl. This occurs when secrets are duplicated across multiple systems, including various repositories, cloud providers, and CI/CD pipelines. The result is a tangled web of inconsistencies that makes tracking and managing secrets a Herculean task.
+
+Updating secrets manually across all these systems is not only tedious but also error-prone. Each time a secret changes, it needs to be updated in every repository, cloud platform, and CI/CD tool where it's used. This process is time-consuming and frustrating, often leading to mistakes that can compromise security.
+
+Moreover, the lack of a centralized system makes it difficult to track where secrets are stored and who has access to them. This complicates security audits and compliance efforts. When a secret is compromised, the process of revoking and rotating it across multiple systems can be slow and disruptive, potentially leading to application downtime.
+
+## ESC Sync with Infrastructure as Code
+
+Using Pulumi building blocks like ESC and IaC, we can create a pattern where secrets and configuration can be centrally defined in ESC, and then automatically synced across the external platforms where they will be utilized.
+
+Consider the following ESC environment, which imports an environment "my-imported-env" and defines a set of secrets and configuration values to be synced:
+
+```yaml
+imports:
+  - my-imported-env@stable
+values:
+  sync:
+    awsSecretsManager:
+      value:
+        myConfigKey: ${my-imported-env-foo}
+        myNestedKey:
+          haha: ${my-imported-env-bar}
+        mySecret: ${my-imported-env-password}
+      name: name-in-secrets-manager
+```
+
+The `sync` block defines the secrets and configuration values to be synced. In this example, we are syncing values to AWS Secrets Manager. The `value` field contains the actual values to be synced, while the `name` field specifies the name of the secret in AWS Secrets Manager.
+
+To automate the synchronization process, we define a Pulumi program that creates an environment, stack, and deployment settings for the target stack. We then define a set of pre-run commands that extract the values from the environment and set them as configuration values in the target stack. The `syncCronSchedule` variable specifies how often the synchronization process should run.
```

Comment:
```suggestion
To automate the synchronization process, we define a Pulumi program that creates an Pulumi ESC environment, Pulumi IaC stack, and Pulumi [deployment](/docs/pulumi-cloud/deployments/) settings for the target stack. We then define a set of pre-run commands that extract the values from the environment and set them as configuration values in the target stack. The `syncCronSchedule` variable specifies how often the synchronization process should run.
```

Reply:
I'm accepting your change but it's a little obnoxious to me how often we're saying Pulumi in that one sentence.